portal/src/main/webapp/js/src/oncoprint/studyview.js

```diff
@@ -31,9 +117,46 @@ $(document).ready(function() {
 			})();
 			$('#outer_loader_img').hide();
 			$('#oncoprint #everything').show();
+                        
+                        var gene_data = response.toJSON();
+                        //get to process data from gene panel
+                        if(genepanelValues.genepaneldata.length>0){
+                            for(var i=0; i < gene_data.length; i++)
+                            {
+                                var geneIndexValue;
+                                var genepanelStableId;
+                                if(genepanelValues.genepaneldata[i] !== undefined)
+                                {
+                                    genepanelStableId = genepanelValues.genepaneldata[i].attr_val;
+                                }
+                                else
+                                {
+                                    genepanelStableId = undefined;
+                                }
+
+                                for(var j= 0; j < config.gene_order.length; j++)
+                                {
+                                    if(genepanelStableId !== undefined)
+                                    {
+                                        geneIndexValue = _.find(genepanelValues.genepanel[genepanelStableId].geneList, function(gene){ return gene === gene_data[i+j*gene_data.length].gene; }); 
```

Comment:
indexing like this ```gene_data[i+j*gene_data.length]``` is not robust to changes in the service - I really think you should collect the data into an organized map and make your adjustments there, then get the values from the map.